src/components/data/data-utils.js

```diff
@@ -939,6 +940,111 @@ export const MultipleEntityEditor = ({ entityType, entityNames, attributeNames,
   ])
 }
 
+export const AddEntityModal = ({ workspaceId: { namespace, name }, entityType, attributeNames, entityTypes, onDismiss, onSuccess }) => {
+  const [entityName, setEntityName] = useState('')
+  const [entityNameInputTouched, setEntityNameInputTouched] = useState(false)
+  const [takenNames, setTakenNames] = useState([])
+
+  // Default all attribute values to empty strings
+  const [attributeValues, setAttributeValues] = useState(_.fromPairs(_.map(attributeName => [attributeName, ''], attributeNames)))
+
+  const [isBusy, setIsBusy] = useState(false)
+
+  // The data table colum heading shows this as `${entityType}_id`, but `${entityType} ID` works better in a screen reader
+  const entityNameLabel = `${entityType} ID`
+
+  const entityNameErrors = validate.single(entityName, {
+    presence: {
+      allowEmpty: false,
+      message: `${entityNameLabel} is required`
+    },
+    format: {
+      pattern: '[a-z0-9_.-]*',
+      flags: 'i',
+      message: `${entityNameLabel} may only contain alphanumeric characters, underscores, dashes, and periods.`
+    },
+    exclusion: {
+      within: takenNames,
+      message: `A ${entityType} with this name already exists`
+    }
+  })
+
+  const createEntity = async () => {
+    setIsBusy(true)
+    try {
+      await Ajax().Workspaces.workspace(namespace, name).createEntity({
+        entityType,
+        name: _.trim(entityName),
+        attributes: _.mapValues(prepareAttributeForUpload, attributeValues)
+      })
+      setIsBusy(false)
+      onDismiss()
+      onSuccess()
+    } catch (err) {
+      setIsBusy(false)
+      if (err.status === 409) {
+        setTakenNames(prevTakenNames => _.uniq(Utils.append(entityName, prevTakenNames)))
+      } else {
+        onDismiss()
+        reportError('Unable to add row.', err)
+      }
+    }
+  }
+
+  return h(Modal, {
+    onDismiss,
+    title: `Add ${entityType}`,
+    okButton: h(ButtonPrimary, {
+      disabled: !!entityNameErrors,
+      tooltip: Utils.summarizeErrors(entityNameErrors),
+      onClick: createEntity
+    }, 'Add')
```

Comment:
```suggestion
    }, ['Add'])
```

See comment above for a similar case.